src/metabase/upload/util.cljc

```diff
@@ -0,0 +1,58 @@
+(ns metabase.upload.util)
+
+(defn unique-name-generator
+  "A fork of [[mbql.u/unique-name-generator]] which accounts for collisions when names are truncated.
+  There is due to a limitation where it does not play well with truncation, see the tests for this function.
```

Comment:
I should reword this, but the limitation is with the `mbql.u` version, not this one.